### PR TITLE
native introduce jsb.window

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: https://github.com/cocos/3d-tasks/issues/15004

### Changelog

* Simulate web context in jsb.window

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/2d/assets/sprite-frame.ts [hunk 1]
```diff
@@ -1241,8 +1241,10 @@ export class SpriteFrame extends Asset {
             this._packable = false;
             return;
         }
+        const engineGlobal = typeof globalThis.jsb !== 'undefined' ? (typeof jsb.window !== 'undefined' ? jsb.window : globalThis) : globalThis;
```
Comment (minggo) [inline cocos/2d/assets/sprite-frame.ts]: Please use a more readable format.

File: cocos/game/splash-screen.ts [hunk 2]
```diff
@@ -123,6 +123,7 @@ export class SplashScreen {
         };
         this._curTime = 0;
 
+        const engineGlobal = typeof globalThis.jsb !== 'undefined' ? (typeof jsb.window !== 'undefined' ? jsb.window : window) : window;
```
Comment (minggo) [inline cocos/game/splash-screen.ts]: Could wrap a function instead?

File: cocos/2d/assets/sprite-frame.ts (hunk 1 above)
Comment (PatriceJiang) [reply cocos/2d/assets/sprite-frame.ts]: the expression has been moved into global-exports.ts

File: cocos/game/splash-screen.ts [hunk 3]
```diff
@@ -135,26 +136,26 @@ export class SplashScreen {
 
             this.initWaterMark();
             const bgPromise = new Promise<void>((resolve, reject) => {
-                this.bgImage = new Image();
-                this.bgImage.onload = () => {
+                this.bgImage = new engineGlobal.Image();
+                (this.bgImage as any).onload = () => {
```
Comment (minggo) [inline cocos/game/splash-screen.ts]: Why need `as any`?

File: cocos/core/platform/debug.ts [hunk 4]
```diff
@@ -152,7 +154,7 @@ export function _resetDebugSetting (mode: DebugMode) {
                 logList.setAttribute('rows', '20');
                 logList.setAttribute('cols', '30');
                 logList.setAttribute('disabled', 'true');
-                const logListStyle = logList.style;
+                const logListStyle = (logList as any).style;
```
Comment (minggo) [inline cocos/core/platform/debug.ts]: Why need `as any`?

File: cocos/core/global-exports.ts [hunk 5]
```diff
@@ -59,3 +59,14 @@ _global.CocosEngine = legacyCC.ENGINE_VERSION = engineVersion;
...
+const engineGlobal: typeof window = typeof globalThis.jsb !== 'undefined' ? (typeof jsb.window !== 'undefined' ? jsb.window : window) : window;
+
+/**
+ * @en
+ * It is jsb.window in native mode, other wise it is window object in web context.
+ * @zh
+ * 原生环境下为 jsb.window, 引擎提为模拟部分 web 环境所提供. Web 环境这个变量是 window 对象.
+ * @deprecated
+ */
+export { engineGlobal };
```
Comment (minggo) [inline cocos/core/global-exports.ts]: The meaning of `engineGlobal` is not very clear. May be use `ccwindow` instead? Or use other more meaningful name. As it is to replace `window`.

File: cocos/asset/assets/image-asset.jsb.ts [hunk 6]
```diff
@@ -58,7 +59,7 @@ function isNativeImage (imageSource: ImageSource): imageSource is (HTMLImageElem
         return false;
     }
 
-    return imageSource instanceof HTMLImageElement || imageSource instanceof HTMLCanvasElement || isImageBitmap(imageSource);
```
Comment (SantyWang) [inline cocos/asset/assets/image-asset.jsb.ts:61]: `isImageBitmap` should be changed to jsbWindow too?

File: cocos/core/platform/debug.ts [hunk 7]
```diff
@@ -59,7 +61,7 @@ function formatString (message?: any, ...optionalParams: any[]) {
  * @param optionalParams - JavaScript objects with which to replace substitution strings within msg.
  * This gives you additional control over the format of the output.
  */
-export function log (message?: any, ...optionalParams: any[]) {
```
Comment (SantyWang) [inline cocos/core/platform/debug.ts]: Should not change these coding style

**Comment by SantyWang** [comment]
Does the usage of window in the mini-game platform also need to be adjusted?

File: cocos/core/platform/debug.ts (hunk 7 above)
Comment (PatriceJiang) [reply cocos/core/platform/debug.ts]: It is set in eslint config file
image

https://eslint.org/docs/latest/rules/space-before-function-parentheses

Comment (PatriceJiang) [reply cocos/core/platform/debug.ts]: I missed

**Comment by PatriceJiang** [comment]
> Does the usage of window in the mini-game platform also need to be adjusted?

Currently no, it should be wrapped in the future

File: cocos/core/platform/debug.ts (hunk 7 above)
Comment (PatriceJiang) [reply cocos/core/platform/debug.ts]: fixed

**Review by minggo — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Pull request merged**